File: java/game.java
//Great performance. Also has some flickers. Also want to try time on this one.
public class game{
    static final int ROWS = 50;
    static final int COLS = 185;
    static int[][] grid = new int[ROWS][COLS];
    static int[][] next = new int[ROWS][COLS];

    public static void main(String[] args) throws Exception{
        initializeGrid();

        while(true){
            printGrid();
            updateGrid();
        }
    }

    static void initializeGrid(){
        for(int i = 0; i < ROWS; i++){
            for(int j = 0; j < COLS; j++){
                if(Math.random() < 0.5){
                    grid[i][j] = 1;
                } else {
                    grid[i][j] = 0;
                }
            }
        }
    }

    static int countNeighbors(int x, int y){
        int sum = 0;
        for(int dx = -1; dx <= 1; dx++){
            for(int dy = -1; dy <= 1; dy++){
                if(!(dx == 0 && dy == 0)){
                    sum += grid[(x + dx + ROWS) % ROWS][(y + dy + COLS) % COLS];
                }
            }
        }
        return sum;
    }

    static void updateGrid(){
        for(int i = 0; i < ROWS; i++){
            for(int j = 0; j < COLS; j++){
                int n = countNeighbors(i, j);
                int cell = grid[i][j];
                if((cell == 1 && (n == 2 || n == 3)) || (cell == 0 && n == 3)){
                    next[i][j] = 1;
                }else{
                    next[i][j] = 0;
                }
            }
        }
        int[][] temp = grid;
        grid = next;
        next = temp;
    }

    static void printGrid(){
        try{
            new ProcessBuilder("clear").inheritIO().start().waitFor();
        }catch(Exception e){
            e.printStackTrace();
        }

        System.out.flush();
        for(int i = 0; i < ROWS; i++){
            StringBuilder line = new StringBuilder();
            for (int j = 0; j < COLS; j++){
                if(grid[i][j] == 1){
                    line.append('*');
                }else{
                    line.append(' ');
                }
            }
            System.out.println(line);
        }
    }
}

File: lua/game.lua
--twice the size and much better performance :O
local rows = 50
local cols = 190
local grid = {}
local next_grid = {}

local function initialize_grid()
	for i = 1, rows do
		grid[i] = {}
		next_grid[i] = {}
		for j = 1, cols do
			grid[i][j] = math.random(0, 1)
			next_grid[i][j] = 0
		end
	end
end

local function count_neighbors(x, y)
	local sum = 0
	for dx = -1, 1 do
		for dy = -1, 1 do
			if not (dx == 0 and dy == 0) then
				local nx = (x + dx - 1) % rows + 1
				local ny = (y + dy - 1) % cols + 1
				if grid[nx][ny] == 1 then
					sum = sum + 1
				end
			end
		end
	end
	return sum
end

local function update_grid()
	for i = 1, rows do
		for j = 1, cols do
			local neighbors = count_neighbors(i, j)
			local state = grid[i][j]
			if state == 0 and neighbors == 3 then
				next_grid[i][j] = 1
			elseif state == 1 and (neighbors < 2 or neighbors > 3) then
				next_grid[i][j] = 0
			else
				next_grid[i][j] = state
			end
		end
	end
	for i = 1, rows do
		for j = 1, cols do
			grid[i][j] = next_grid[i][j]
		end
	end
end

local function print_grid()
	for i = 1, rows do
		for j = 1, cols do
			io.write(grid[i][j] == 1 and "*" or " ")
		end
		io.write("\n")
	end
end

math.randomseed(os.time())
initialize_grid()

while true do
	print_grid()
	update_grid()
end
